### PR TITLE
preprocess_frames! function

### LINKED ISSUE #260: Preprocessing of Frames for `get_javis_frame`
**Is your feature request related to a problem? Please explain.**

Currently, it is difficult to return a single frame of an animation to a user in an easy and performant manner.

**Describe the solution you'd like**

Javis should provide functionality with the `get_javis_frame` function to preprocess a chosen frame and deliver it back to the user quickly. @Wikunia, I am not sure exactly how we would go about implementing this but you thought this should be relatively easy/straightforward to fix.

**Additional context**
Linked to issue #92 for allowing a user to easily get the frame they want.

### PR DESCRIPTION
**Pull request by Wikunia**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [ ] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [x] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [x] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [x] Did I properly add test dependencies to the `test` directory (if applicable)?
- [x] Did I check relevant tutorials that may be affected by changes in this PR?
- [x] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #260 

**How did you address these issues with this PR? What methods did you use?**
I've created a `preprocess_frames!` function which computes the correct frames for all objects and actions. It can also be used for #214 at one point. The function can be used when neither the live preview should be shown nor a gif/mp4 should be rendered but instead a single frame should be computed with `get_javis_frame`.

Additionally I've updated the unit tests to use this functionality.

### CODE REVIEW

File: src/Javis.jl [hunk 1]
```diff
@@ -119,6 +119,37 @@ function projection(p::Point, l::Line)
     return c * v + o
 end
 
+"""
+    preprocess_frames!(objects::Vector{<:AbstractObject})
+
+Computes the frames for each object and action based on the user defined frames that the
+user can provide like `Rel` and `:same`.
```
Comment (TheCedarPrince) [inline src/Javis.jl:126]: Only thing here is to change `Rel` to match the `RFrame` syntax. Thanks!

Comment (TheCedarPrince) [inline src/Javis.jl:126]: Could you add to the docstring about why this preprocess is needed for the `get_javis_frame` method? I understand that it defines CURRENT_OBJECT, but I had to do a bit of poking around to figure out what exactly it was doing.

**Review by TheCedarPrince — CHANGES_REQUESTED** [review]
Only two docstring comments but otherwise, LGTM. Great work! Do you think this would be an appropriate time to also export both `get_javis_frame` and `preprocess_frames!`?

File: src/Javis.jl (hunk 1 above)
Comment (Wikunia) [reply src/Javis.jl]: I've added a docstring. `CURRENT_OBJECT` is not that important I think. It can maybe get removed there at some point.

**Review by TheCedarPrince — APPROVED** [review]
